junit-platform-launcher/src/main/java/org/junit/platform/launcher/TestIdentifier.java

```diff
@@ -61,18 +63,20 @@ public static TestIdentifier from(TestDescriptor testDescriptor) {
 		boolean container = !test || !testDescriptor.getChildren().isEmpty();
 		Optional<String> parentId = testDescriptor.getParent().map(
 			parentDescriptor -> parentDescriptor.getUniqueId().toString());
-		return new TestIdentifier(uniqueId, displayName, source, tags, test, container, parentId);
+		final DefaultLegacyReportingInfo legacyReportingInfo = DefaultLegacyReportingInfo.from(testDescriptor.getLegacyReportingInfo());
```

Comment:
I don't think the `final` keyword is need here either, as the variable is used immediately on the next line.